crates/noirc_frontend/src/graph/mod.rs

```diff
@@ -97,7 +100,23 @@ impl CrateGraph {
         }
 
         let data = CrateData { root_file_id: file_id, crate_type, dependencies: Vec::new() };
-        let crate_id = CrateId(self.arena.len());
+        let crate_id = CrateId::Crate(self.arena.len());
+        let prev = self.arena.insert(crate_id, data);
+        assert!(prev.is_none());
```

Comment:
Does this insert fix anything? I'm curious why we didn't insert before this change

Reply:
It did, you are just looking at the diff wrong. Most of the code was duplicated except creating a `CrateId::Stdlib(self.arena.len());` instead of `CrateId::Crate(self.arena.len());`